### PR TITLE
UserInfo parameter pull request.

### PR DESCRIPTION
**Pull request by WildDylan**
for #36 
re fork the project, re download xcode, every thing is ok.

---
This change is [Reviewable](https://reviewable.io/reviews/devxoul/urlnavigator/37)

### CODE REVIEW

File: Sources/URLNavigator.swift [hunk 1]
```diff
@@ -179,6 +180,42 @@ open class URLNavigator {
...
+  /// - parameter from: The navigation controller which is used to push a view controller. Use application's top-most
+  ///     view controller if `nil` is specified. `nil` by default.
+  /// - parameter animated: Whether animates view controller transition or not. `true` by default.
+  ///
+  /// - returns: The pushed view controller. Returns `nil` if there's no matching view controller or failed to push
+  ///            a view controller.
+  @discardableResult
+  open func push(
+    _ url: URLConvertible,
+    _ userInfo: [AnyHashable: Any]?,
```
Comment (devxoul) [inline Sources/URLNavigator.swift]: Why not just adding `userInfo` parameter with default value (`nil`) to existing `push()` and `present()` function?

File: Tests/URLNavigatorPublicTests.swift [hunk 2]
```diff
@@ -77,6 +77,21 @@ class URLNavigatorPublicTests: XCTestCase {
     XCTAssertEqual(navigationController.viewControllers.count, 2)
   }
 
+  func testPushURLWithUserInfo_URLNavigable() {
```
Comment (devxoul) [inline Tests/URLNavigatorPublicTests.swift]: Could you rename this with `testPushURL_userInfo`?

File: Tests/URLNavigatorPublicTests.swift [hunk 3]
```diff
@@ -108,6 +123,19 @@ class URLNavigatorPublicTests: XCTestCase {
     XCTAssertNil(viewController)
   }
 
+  func testPresentURLWithUserInfo_URLOpenHandler() {
```
Comment (devxoul) [inline Tests/URLNavigatorPublicTests.swift]: Could you rename this with `testPresentURL_userInfo`?

File: Sources/URLNavigator.swift [hunk 4]
```diff
@@ -117,11 +117,12 @@ open class URLNavigator {
   /// Returns a matched view controller from a specified URL.
   ///
   /// - parameter url: The URL to find view controllers.
+  /// - parameter userInfo: The user extra parameters you want add.
   /// - returns: A match view controller or `nil` if not matched.
-  open func viewController(for url: URLConvertible) -> UIViewController? {
+  open func viewController(for url: URLConvertible, with userInfo: [AnyHashable: Any]? = nil) -> UIViewController? {
```
Comment (devxoul) [inline Sources/URLNavigator.swift]: I think we don't need the parameter name `with`. It is OK to use just `userInfo` :smile:

Comment (WildDylan) [inline Sources/URLNavigator.swift]: U are right. 
In my mind, Just want to let the code like:

```swift
viewController(for: "", with: infos)
```

File: Sources/URLNavigator.swift (hunk 1 above)
Comment (WildDylan) [inline Sources/URLNavigator.swift]: Latest commit remove the added function.

**Comment by devxoul** [comment]
@WildDylan, This is awesome. Thank you very much for your contribution 👍🎉 I'll make a new release soon :)

**Comment by WildDylan** [comment]
Pure awesomeness, thanks a lot

**Comment by devxoul** [comment]
This PR has been shipped with [1.1.0](https://github.com/devxoul/URLNavigator/releases/tag/1.1.0) release. Thanks again for you work :tada:

**Comment by WildDylan** [comment]
Pure awesomeness!

**Pull request merged**